degiro_connector/trading/actions/action_confirm_order.py

```diff
@@ -140,9 +140,14 @@ def confirm_order(
         except requests.HTTPError as e:
             logger.fatal(response_raw.status_code)
             logger.fatal(response_raw.text)
+            response_dict = response_raw.json()
```

Comment:
If at this point you are sure to have a `response_dict` I suggest returning it directly and leave the rest of the code as it is.